test/unit/stateView/document/query/validateIndexedFields.spec.js

```diff
@@ -0,0 +1,179 @@
+const validateIndexedFields = require('../../../../../lib/stateView/document/query/validateIndexedFields');
+
+describe('validateIndexedFields', () => {
+  let indexedFields;
+
+  beforeEach(() => {
+    indexedFields = [
+      [{ $userId: 'asc' }, { firstName: 'desc' }],
+      [{ $userId: 'asc' }, { lastName: 'desc' }, { secondName: 'asc' }],
+      [{ $id: 'asc' }],
```

Comment:
We don't allow to create indices with $id and $userId. Only if they are part of compound index

Reply:
Yes. We add them in `getIndexedFieldsFromDocumentSchema`. These fields are from that function. Or you want me to add them in another place?

Reply:
Nope, it's ok. Then let's add `[{ $userId: 'asc' }]` too

Reply:
Got it